include/tvm/runtime/packed_func.h

```diff
@@ -442,13 +447,17 @@ class TVMPODValue_ {
  protected:
   friend class TVMArgsSetter;
   friend class TVMRetValue;
+  friend class TVMMovableArgValue_;
   TVMPODValue_() : type_code_(kTVMNullptr) {}
-  TVMPODValue_(TVMValue value, int type_code) : value_(value), type_code_(type_code) {}
+  TVMPODValue_(TVMValue value, int type_code, const std::string& prefix_message = "")
+      : value_(value), type_code_(type_code), prefix_message_(prefix_message) {}
 
   /*! \brief The value */
   TVMValue value_;
   /*! \brief the type code */
   int type_code_;
+  /*! \brief Message to prefix any error message with */
+  std::string prefix_message_;
```

Comment:
Adding String to each of the argument is a big overhead. We certainly don't want to construct such an error message for each function call even if no error is presented.

Instead, it might be good to think about alternatives. E.g. trying to be able to catch the errors in FFI functions and add name to the trace.

Reply:
> Instead, it might be good to think about alternatives. E.g. trying to be able to catch the errors in FFI functions and add name to the trace.

This code gives error messages when the arguments cannot be converted correctly. It prints which argument does not match. I don't see how catching errors at the FFI can tell us which argument was incorrect.


The only other approach I can think of is manually converting each argument before we call the function. This would be a much larger change and would probably involve some template magic.

Reply:
I understand what this code does, want to point out the overhead in the current approach is too large that we might want to think about alternatives :)

The efficiency and clean part of the packedfunc mechanism really matters here that we might not want to include the feature because of the eager construction overhead.

I am not suggesting the FFI catching is the only way, but I do think more thoughts need to put into the process, for example, to have the error message report the i-th argument when things went wrong, while catching will show the name of the function.

Reply:
Happy to chat more, i think there could be a better approach here